src/main/resources/liquibase/crop_changelog/15_4_0.xml

```diff
@@ -166,4 +166,26 @@
 			ALTER TABLE search_request CHANGE COLUMN parameters parameters MEDIUMTEXT NOT NULL ;
 		</sql>
 	</changeSet>
+
+	<changeSet author="mderamos" id="v15.4.0-7">
+		<preConditions onFail="MARK_RAN">
+            <and>
+                <not>
+                    <foreignKeyConstraintExists foreignKeyName="germplasm_study_source_gid_fk"></foreignKeyConstraintExists>
+                </not>
+            </and>
+		</preConditions>
+		<comment>Delete germplasm_study_source entries with no germplsm</comment>
+		<sql dbms="mysql" splitStatements="true">
+			CREATE TEMPORARY TABLE non_existent_gids
```

Comment:
I don't see the need to create a temporary table. 
Can't we just a single script like:
```
DELETE FROM germplasm_study_source 
WHERE NOT EXISTS (SELECT 1 FROM germplsm g WHERE g.gid = germplasm_study_source.gid);
```